client/main.coffee

```diff
@@ -47,10 +47,18 @@ Player = ReactClass
       'player': true
       'selected': @props.selected
 
-    (div {className:classes, onClick: @selectPlayer}, [
-      (span {className:"name" }, @props.player.name )
-      (span {className:"score"}, @props.player.score)
-    ])
+    # coffeescript ignores newlines. human taste buds do not. 
+    div 
+      className: classes
+      , onClick: @selectPlayer
+    , [ 
+      span 
+        className: "name"
+      , @props.player.name
+      span 
+        className: "score"
+      , @props.player.score
+    ]
```

Comment:
There’s plenty of merit to your style choice. For me there are plenty of reasons
to love leading commas.
You get to read code vertically. Full stop.You can comment out individual arguments etc.Markup doesn’t follow DOM. It’s better. (pure opinion)Maximum amounts of air makes it look easily penetrated... If you’re ok with
whitespace sensitive…
In a word: Yes.
(div {className:classes, onClick: @selectPlayer}, [ > - (span {className:”name”
}, @ props.player.name [http://props.player.name/] ) > - (span {className:”score”}, @props.player.score) > - ]) > + # coffeescript
ignores newlines. human taste buds do not. > + div > + className: classes > + ,
onClick: @selectPlayer > + , [ > + span > + className: “name” > + , @ props.player.name [http://props.player.name/] > + span > + className: “score” > + , @props.player.score > + ]
When skimming the leading comma style you see the div on a new row. If looking
for the span you just hop along. You have a point with the () especially since
an ide can light up the opposing ). Just don’t like to rely on that.
By the way who agrees with this? https://speakerdeck.com/vjeux/react-css-in-js
[https://speakerdeck.com/vjeux/react-css-in-js]
Here’s one that manage to sneak by you :)
Reactor.onBeforeAction -> if not Meteor.userId() @redirect ‘login’ else @next()
, except: [’login’, ‘signup’]
This one should fry ones brain and the comma alerts you to the weirdness
instantly. Prefer unless to if not by the way.
Design is opinion. Give it some thought. Or not.
Either way, reactor looks very nice. Be well




On Tue, Mar 3, 2015 at 12:19 AM, Chet Corcos< notifications@github.com [notifications@github.com] > wrote:
In client/main.coffee
[https://github.com/ccorcos/meteor-react-leaderboard/pull/4#discussion_r25634197] :

> - (div {className:classes, onClick: @selectPlayer}, [ > - (span
{className:"name" }, @ props.player.name [http://props.player.name] ) > - (span {className:"score"}, @props.player.score) > - ]) > + # coffeescript
ignores newlines. human taste buds do not. > + div > + className: classes > + ,
onClick: @selectPlayer > + , [ > + span > + className: "name" > + , @ props.player.name [http://props.player.name] > + span > + className: "score" > + , @props.player.score > + ]You think this is more readable??

—
Reply to this email directly or view it on GitHub
[https://github.com/ccorcos/meteor-react-leaderboard/pull/4/files#r25634197] .




ELIAS MOOSMAN




UAE +971566744336USA +19177250603

HAPPY VITAMINS - HTTP://WHOGOTHAPPY.COM [HTTP://WHOGOTHAPPY.COM/]


TAPMATE - EXPRESSION -GOT THE APP? HTTP://TAPMATE.YOUIEST.COM [HTTP://TAPMATE.YOUIEST.COM/]